catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -479,20 +481,44 @@ public DeleteResponse delete(DeleteRequest deleteRequest) throws IngestException
             return new DeleteResponseImpl(deleteRequest, null, deletedMetacards);
         }
 
-        /* 1. Query first for the records */
+        if (identifiers.size() <= DEFAULT_PAGING_SIZE) {
+            deleteListOfMetacards(deletedMetacards, identifiers, attributeName);
+        } else {
+            List<? extends Serializable> identifierPaged = null;
+            int currPagingSize = 0;
+
+            for (currPagingSize = DEFAULT_PAGING_SIZE; currPagingSize < identifiers.size(); currPagingSize += DEFAULT_PAGING_SIZE) {
```

Comment:
This is fine, but you might consider partitioning the list with http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Lists.html#partition(java.util.List,%20int) and iterating that list of lists.